drivers/frequency/ad9528/ad9528.c

```diff
@@ -380,12 +380,13 @@ int32_t ad9528_setup(struct ad9528_dev **device,
 		return ret;
 
 	/* GPIO */
-	if(init_param.hw_reset_en) {
-		ret = gpio_get(&dev->gpio_resetb, &init_param.gpio_resetb);
-		if (ret < 0)
-			return ret;
+	ret = gpio_get_optional(&dev->gpio_resetb, init_param.gpio_resetb);
+	if (ret < 0)
+		return ret;
+
+	if (dev->gpio_resetb)
 		ad9528_reset(dev);
-	}
+
 	ret = ad9528_spi_write_n(dev,
```

Comment:
I like the new gpio_get_optional !
However, this soft reset (command over SPI) should be done only if no HW reset was done. It doesn't harm to also do it when a hard reset was done, but it doesn't make sense. So I simply suggest to move it in an else branch of the if (dev->gpio_resetb) condition. ;)

Reply:
Without some additional testing, I would still keep the software reset too.